Dockerfile

```diff
@@ -190,7 +190,7 @@ ENV AIRFLOW_INSTALL_USER_FLAG="--user"
 ENV AIRFLOW_INSTALL_EDITABLE_FLAG=""
 
 # Upgrade to specific PIP version
-RUN pip install --upgrade "pip==${AIRFLOW_PIP_VERSION}"
+RUN pip install --upgrade "pip~=${AIRFLOW_PIP_VERSION}"
```

Comment:
https://www.python.org/dev/peps/pep-0440/#compatible-release

`pip~=21.0.1` is equivalent to `piip>=21.0.1, ==21.0.*`

Which from the PR description is not what I think you intended.

Reply:
Well that what I actually intended :) ~= is defined as "compatible" release (at least that's the name PIP definition uses) - and I think this makes sense to always increment patchlevel automatically but then make a deliberate decisions about major level increases (looking at PIP history this is a reasonable) - even if it means that only new features are added. 

So in short `>=21.0.1, ==21.0.*` is exactly what I wanted to achieve.

Reply:
Cool. I thought you wanted 21.* when you said "21* series release". LGTM